### PR TITLE
Document that Kubernetes enables portability.

### PR DESCRIPTION
**Pull request by bgrant0607**

### CODE REVIEW

File: docs/whatisk8s.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ Kubernetes satisfies a number of common needs of applications running in product
 * [support for introspection and debugging](user-guide/introspection-and-debugging.md), and
 * [identity and authorization](admin/authorization.md).
 
-This provides the simplicity of Platform as a Service (PaaS) with the flexibility of Infrastructure as a Service (IaaS).
+This provides the simplicity of Platform as a Service (PaaS) with the flexibility of Infrastructure as a Service (IaaS), an enables portability across infrastructure providers.
```
Comment (quinton-hoole) [inline docs/whatisk8s.md]: typo. an -> and

Comment (quinton-hoole) [inline docs/whatisk8s.md]: Also, I think that we need to provide some more detail here, or somewhere else in this doc (or perhaps a separate "how do I make sure that my application is portable" doc).  The key question that needs answering is: if I develop my application, and it's Kubernetes deployment config on one cluster, can I trivially deploy it on any other Kubernetes cluster, and have it work correctly?  If not "any" other cluster, what are the properties of my app and the other cluster that are required for my app to run successfully on the other cluster?  Off the top of my head:

1. Any K8s addons upon which my app depends must be installed on both clusters.  How do I determine which addons I depend on?  To what extent can I replace one addon with an alternative, equivalent one.
2. What is the dependency on the node OS (this is more a docker question).
3. What is the dependency on the cloud provider?
etc...

Comment (bgrant0607) [inline docs/whatisk8s.md]: This is an overview, which is already getting long.

I agree we need a separate doc.

**Comment by bgrant0607** [comment]
Fixed typo. Will squash after lgtm.

**Comment by bgrant0607** [comment]
Weakened portability language slightly, due to ongoing discussion. Now this is just minor edits. Squashed. PTAL.

**Pull request merged**